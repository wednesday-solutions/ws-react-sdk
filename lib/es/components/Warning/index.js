import{_ as e,a as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"../_rollup-plugin-inject-process-env-1259e43d.js";import t from"react";import{c as n}from"../colors-93a064d8.js";import{f as o}from"../fonts-ca6864a0.js";import{m as a}from"../media-ee9cbc51.js";import m from"styled-components";import i from"../T/index.js";import"react-intl";import"prop-types";import"../If/index.js";var s,p=m(i)(s||(s=e(["\n  && {\n    color: ",";\n    margin: 2rem 0rem 0rem 0rem;\n    ",";\n\n    // mobile\n    ",";\n    \n    // tablet\n    ",";\n\n    // desktop\n    ",";\n"])),n.pawlyBlue,o.weights.fw600(),a.tablet.max(o.size.large()),a.tablet.min(o.size.xRegular()),a.desktop.min("\n      ".concat(o.size.extraLarge(),";\n      ").concat(o.weights.bold(),";\n    ")));function l(e){return t.createElement(p,r({"data-testid":"warning"},e))}l.propTypes={};export default l;
