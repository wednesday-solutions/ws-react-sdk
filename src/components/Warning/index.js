/**
 *
 * Warning
 *
 */

import React from 'react';
import { media, fonts, colors } from '@themes';
import styled from 'styled-components';
import T from '../T';
const StyledWarning = styled(T)`
  && {
    color: ${colors.appBlue};
    margin: 2rem 0rem 0rem 0rem;
    ${fonts.weights.fw600()};

    // mobile
    ${media.tablet.max(fonts.size.large())};
    
    // tablet
    ${media.tablet.min(fonts.size.xRegular())};

    // desktop
    ${media.desktop.min(`
      ${fonts.size.extraLarge()};
      ${fonts.weights.bold()};
    `)};
`;
function Warning(props) {
  return <StyledWarning data-testid="warning" {...props} />;
}

Warning.propTypes = {};

export default Warning;
