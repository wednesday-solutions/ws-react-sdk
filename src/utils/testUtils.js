import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
