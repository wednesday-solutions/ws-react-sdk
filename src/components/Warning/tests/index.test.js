/**
 *
 * Tests for Warning
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import Warning from '../index';

describe('<Warning />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Warning />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Warning component', () => {
    const { getAllByTestId } = renderWithIntl(<Warning />);
    expect(getAllByTestId('warning').length).toBe(1);
  });
});
