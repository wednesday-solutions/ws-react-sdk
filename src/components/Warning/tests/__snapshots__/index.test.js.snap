// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Warning /> should render and match the snapshot 1`] = `
<body>
  <div>
    <p
      class="sc-bdfBwQ bZojWc sc-gsTCUz cWOkuF"
      data-testid="warning"
    />
  </div>
</body>
`;
